### DISCUSSION TITLE
uma_authorization role usage

### DISCUSSION THREAD

**Discussion by syalioune** [keycloak/keycloak#17529] [Q&A]
Hello,

I am reviewing the configuration of one keycloak instance and I'm wondering about the impact of removing the `uma_authorization` role or at least removing it from the default role (**please note that I'm not using authorization services**).
There are no references to this role in the documentation.

From what I can tell, it was introduced by [KEYCLOAK-2753](https://github.com/keycloak/keycloak/blob/086c29112a70a020d3290f43f82f4bb6eda4a52f/services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java#L89-L95) and was used to scope RPT requests

```java
KeycloakEvaluationContext evaluationContext = new KeycloakEvaluationContext(this.authorization.getKeycloakSession());
        KeycloakIdentity identity = (KeycloakIdentity) evaluationContext.getIdentity();

        if (!identity.hasRole("uma_authorization")) {
            throw new ErrorResponseException(OAuthErrorException.INVALID_SCOPE, "Requires uma_authorization scope.", Status.FORBIDDEN);
        }
```

That part is not present in the current codebase and I cannot find any reference to `uma_authorization` or `AUTHZ_UMA_AUTHORIZATION` other than in tests (mainly arquillian), CLI or constant file.

Is that `uma_authorization`  role still relevant ?

**Comment by syalioune** [comment]
Anybody have the answers ?

**Reply by si90** [reply]
A little bit late, but regarding [redhat](https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.1/html/authorization_services_guide/service_overview#service_authorization_api) `uma_authorization` provides the an authorization API token (AAT) which `"An AAT enables a client application to query the server for user permissions."`

**Reply by syalioune** [reply]
Thanks @si90. This is in line with what I gathered earlier. That is `uma_authorization` was or still is very tied to authorization services.
It does not seem to be required if one doesn't use those services.
